torch/distributions/exp_family.py

```diff
@@ -61,5 +63,5 @@ def entropy(self):
         gradients = torch.autograd.grad(lg_normal.sum(), nparams, create_graph=True)
         result += lg_normal
         for np, g in zip(nparams, gradients):
-            result -= (np * g).reshape(self._batch_shape + (-1,)).sum(-1)
+            result -= cast(float, (np * g).reshape(self._batch_shape + (-1,)).sum(-1))
```

Comment:
this isn't great either, why can't -= take in floats in the typing signature?

Reply:
@Skylion007: Yeah, I'm agreeing with all your comments here.

The question is this - there are many places we pass tensor scalars (which look a whole lot like floats or ints but aren't actually floats or ints) to unsuspecting other places that have no knowledge of tensors at all and just want a regular old number.

So there _is_ no "scalar Tensor" type. It's just a state that a tensor class can be where it easily interoperates with arithmetic.

I'm not exactly sure what the answer is, because I don't think changing the signatures of functions that are doing simply arithmetic ops to take and return Tensors will be a good general solution.